pkg/plugins/gateway/gateway.go

```diff
@@ -133,7 +138,22 @@ func (s *Server) selectTargetPod(ctx *types.RoutingContext, pods types.PodList)
 	if err != nil {
 		return "", err
 	}
-	return router.Route(ctx, pods)
+
+	if pods.Len() == 0 {
+		return "", fmt.Errorf("no pods to forward request")
+	}
+	readyPods := utils.FilterRoutablePods(pods.All())
+	if len(readyPods) == 0 {
+		return "", fmt.Errorf("no ready pods available for fallback")
+	}
+	if len(readyPods) == 1 {
```

Comment:
do you mean if `len(readyPods) >= 1` or `==`?

Reply:
`==`.

If there is only one ready pod, then no need to running routing algorithm. Just return that ready pod.